src/shared/codelens/localLambdaRunner.ts

```diff
@@ -323,6 +323,47 @@ export class LocalLambdaRunner {
             )
         }
     }
+} // end class LocalLambdaRunner
+
+export async function getRuntimeForLambda(params: {
+    handlerName: string,
+    templatePath: string,
+}): Promise<string> {
+    const samTemplateData: CloudFormation.Template = await CloudFormation.load(params.templatePath)
+    if (!samTemplateData.Resources) {
+        throw new Error(
+            `Please specify Resource for '${params.handlerName}' Lambda in SAM template: '${params.templatePath}'`
+        )
+    }
+    const runtimes = new Set<string>()
+    for (const resourceKey in samTemplateData.Resources) {
```

Comment:
Is `for (const resourceKey in samTemplateData.Resources)` functionally equivalent to working with `Object.getOwnPropertyNames(samTemplateData.Resources)`? This is how detectLocalLambdas.ts and a few other places seem to process the `Resources` field.

Reply:
[for in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) and [for of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) iterate over the enumerable properties whereas [Object.getOwnPropertyNames](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames) 

> returns an array of all properties (including non-enumerable properties except for those which use Symbol) found directly in a given object

In this case the result is likely the same, however; generally one does not expect to iterate over non-enumerable properties.